src/nprintml/label/aggregator/pcap.py

```diff
@@ -1,10 +1,17 @@
+import itertools
 import pathlib
 
+import numpy as np
 import pandas as pd
 
+from nprintml.util import storeresults
+
 from . import LabelAggregator, AggregationLengthError, AggregationPathError
 
 
+NPT_DTYPE = np.dtype('int8')
```

Comment:
In my quick consideration this is a reasonable optimization of NumPy/Pandas's handling of nPrint data :shrug:

Reply:
This is a totally reasonable assumption. Unfortunately, we have a few features that could extend beyond `int8`, such as the relative timestamps of each incoming packet (`-R` in nPrint). Should we just look for this column and if we don't find it use `int8`?

Reply:
Ah that makes sense.

What data type is appropriate for the relative timestamps? (Are they just unbounded?) I'm tempted to just choose a single dtype that is universally appropriate, (though depending on the implications of that we can certainly make it check).

Reply:
This is now resolved.

At least for now, we seem to do fine by treating them as default `int` at first (generally int64) → float (due to NaN) → int8+ (either int8 or larger as allowed by downcast).

I've checked that this does indeed save a lot of RAM and it's about as speedy as before.